### DISCUSSION TITLE
Change class User to old class name `FirebaseUser`

### DISCUSSION THREAD

**Discussion by doiya46** [firebase/flutterfire#3238] [Feature request]
class `User` is general for almost project, but when use with `firebase_auth`, the old version name is `FirebaseUser`. Now almost file we need to refactor like this `import 'package:firebase_auth/firebase_auth.dart' as firebase_auth;` and use `firebase_auth.User`, it's a bad name. And almost class need to change like

```dart
firebase_auth.FirebaseAuth.instance.signOut
firebase_auth.GoogleAuthProvider
firebase_auth.AuthCredential
.....
....
```

**Comment by ToniTornado** [comment]
I refactored my `User` class to be an `AppUser` to work around this.
